cookbooks/cdo-varnish/templates/default/accept-language.vcl.erb

```diff
@@ -0,0 +1,188 @@
+C{
+
+/* ------------------------------------------------------------------ */
+/* THIS FILE IS AUTOMATICALLY GENERATED BY ./gen_vcl.pl. DO NOT EDIT. */
+
+/*
+ * Accept-language header normalization
+ *
+ * - Parses client Accept-Language HTTP header
+ * - Tries to find the best match with the supported languages
+ * - Writes the best match as req.http.X-Varnish-Accept-Language
+ *
+ * First version: Cosimo, 21/Jan/2010
+ * Last update:   Cosimo, 03/Nov/2011
+ *
+ * http://github.com/cosimo/varnish-accept-language
+ *
+ */
+
+#include <ctype.h>  /* isupper */
+#include <stdio.h>
+#include <stdlib.h> /* qsort */
+#include <string.h>
+
+#define DEFAULT_LANGUAGE "en"
+#define SUPPORTED_LANGUAGES ":<%= node['cdo-apps']['i18n']['languages'].to_hash.keys.join(':') %>:"
+
+#define vcl_string char
+#define LANG_LIST_SIZE 16
+#define HDR_MAXLEN 256
+#define LANG_MAXLEN 8
+#define RETURN_LANG(x) { \
```

Comment:
I think RETURN_LANG is incorrect in the case when x is longer than LANG_MAXLEN, because strncpy won't add a terminating null in that case, resulting in a non-null-terminated string. 

(I realize this a pre-existing issue but we might want to consider whether it needs to be fixed because it might lead to a varnish crash.)

From http://man7.org/linux/man-pages/man3/strcpy.3.html:  "If there is no null byte among the first n  bytes of src, the string placed in dest will not be null-terminated."